project/frontend/src/pages/HomePage.js

```diff
@@ -4,23 +4,44 @@ import React, { useState } from 'react';
 
 import Modal from '../components/Modal.js';
 import PreferenceForm from '../components/PreferenceForm.js';
+import LocationFinder from '../components/LocationFinder';
 
 function HomePage(props) {
   const [isPreferenceFormOpen, setIsPreferenceFormOpen] = useState(false);
+  const [isLocationFormOpen, setIsLocationFormOpen] = useState(false);
 
   const togglePreferenceForm = () => {
     setIsPreferenceFormOpen((prev) => !prev);
   };
 
+  const toggleLocationForm = () => {
+    setIsLocationFormOpen((prev) => !prev);
+  };
+
+  const handleClick = () => {
+    togglePreferenceForm();
+    toggleLocationForm();
+  };
+
+  const handleData = (data) => {
+    if (data) {
+      props.history.push(data);
+    }
+  };
```

Comment:
Instead of passing data from LocationForm -> HomePage through the `props.history` which is kind of an anti-pattern since props are generally supposed to be immutable but `props.history` is a rare exception to that rule, I think you should declare a another state variable on the HomePage component:

```
const [location, setLocation] = useState({});
```

And then simply pass the `setLocation` function to the `LocationForm` component in a prop so that `LocationForm` can directly update `HomePage`'s state.